### ISSUE TITLE
Poor metrics performance from cri-dockerd (rancher desktop 1.2.x with dockerd)

### ISSUE THREAD

**Issue by chancez** [rancher-sandbox/rancher-desktop#2012]
### Actual Behavior

cri-dockerd metrics performance is much worse than with dockershim. See https://github.com/Mirantis/cri-dockerd/issues/15 https://github.com/rancher/rke/issues/2716 and https://github.com/rancher/rke/issues/2709

### Steps to Reproduce

- `LIMA_HOME="$HOME/Library/Application Support/rancher-desktop/lima" "/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl" shell 0`
- `sudo apk add curl jq`
- `TOKEN="$(cat $(docker inspect "$(docker ps | grep metrics-server | grep -v POD | awk '{print $1}')" | jq '.[0].Mounts[] | select(.Destination=="/var/run/secrets/kubernetes.io/serviceaccount").Source'  -r)/token)"`
- `time curl https://192.168.192.50:10250/stats/summary?only_cpu_and_memory=true -k -v -H "Authorization: Bearer ${TOKEN}"`

### Result

Takes 10+ seconds to get a response, whereas with 1.1.1 it takes less than 10ms
```
...
* Connection #0 to host 192.168.192.50 left intact
}real	0m 11.70s
user	0m 0.01s
sys	0m 0.00s
```

### Expected Behavior

Querying metrics is fast.

### Rancher Desktop Version

1.2.x

### Rancher Desktop K8s Version

v1.22.7

### Which container runtime are you using?

moby (docker cli)

### What operating system are you using?

macOS

### Operating System / Build Version

Darwin dune.local 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000 arm64

### What CPU architecture are you using?

arm64 (Apple Silicon)

**Comment by chancez** [comment]
https://github.com/Mirantis/cri-dockerd/pull/38 seems to fix the issue. Replacing the binary and running `/etc/init.d/cri-dockerd restart` and curling the endpoint returns immediately.

**Comment by gunamata** [comment]
will be fixed by #2022
